### PR TITLE
matrix-mock-request to 2.0.1

### PR DESCRIPTION
**Pull request by kerryarchibald**
https://github.com/matrix-org/matrix-js-sdk/issues/2353

with https://github.com/matrix-org/matrix-mock-request/pull/14

- update matrix-mock-request to v2.0.1
- fix issue with tests logging after completing that surfaced after mock-req update. Track offending timeouts from mock clients and clear in afterEach
- upgrade new ts TestClient for typed mock backend

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by turt2live — APPROVED** [review]

File: src/client.ts [hunk 1]
```diff
@@ -590,13 +590,9 @@ export interface IRequestMsisdnTokenResponse extends IRequestTokenResponse {
     intl_fmt: string;
 }
 
-interface IUploadKeysRequest {
+export interface IUploadKeysRequest {
```
Comment (kerryarchibald) [inline src/client.ts:593]: This type was incorrect, the usage here matches the new type [psf-874/test-messageactionbar](https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/crypto/index.ts#L2003-L2026)

Comment (kerryarchibald) [reply src/client.ts:593]: Added breaking change label

Comment (turt2live) [reply src/client.ts:593]: I don't think this is a breaking change? The interface wasn't exported before, so it'd just be a regular bug fix.

**Review by turt2live — APPROVED** [review]
otherwise lgtm

File: src/client.ts (hunk 1 above)
Comment (kerryarchibald) [reply src/client.ts:593]: I wasn't really sure here so erred on the safe side. Removed the label.

**Pull request merged**